### ISSUE TITLE
Re-add a transform's usedHelpers metadata

### ISSUE THREAD

**Issue by philipwalton** [babel/babel#6835]
**Feature Request:**

In https://github.com/babel/babel/pull/6343 the `usedHelpers` and `metadata.usedHelpers` objects were removed. It would be great to get that info back when doing transforms.

The use case is if babel is transforming lots of individual files for a library or application, and you want all of those files to depend on a single, global, `babelHelpers` object, but you want that object to **only include the helpers needed** from the transforms run.

If the usedHelpers object were exposed, then other build tools could consume this data and automatically generate the minimal set of helpers needed.

**Comment by loganfsmyth** [comment]
Have you explored using `transform-runtime`? In Babel 6 it has the unfortunate side-effect of pulling in helpers that reference `core-js`, but since this issue is about Babel 7, the `useBuiltins` option should already inject nice import statements for the helpers that are used.

**Comment by philipwalton** [comment]
I have not. @hzoo just pointed me to the discussion in #5699, so I'll look into whether or not that solves my use case.

I'm certainly not tied to the specifics of `usedHelpers` in particular, just to the idea of making that info available in some way to users.

**Comment by Andarist** [comment]
You might also look into the "module" output type of `buildExternalHelpers` tool. You can check out https://github.com/rollup/rollup-plugin-babel/pull/170 how the problem of `usedHelpers` absence was solved.

That being said I had also some other problem with the lack of it when doing - https://github.com/rollup/rollup-plugin-babel/pull/177 . It's unreasonably hard to check if the helper was already inserted or not.

**Comment by loganfsmyth** [comment]
Yeah to be clear, I 100% want there to be better hooks to do this kind of thing, I just removed `usedHelpers` because it felt like a very poorly defined/documented list to begin with. It seems like what you're describing is achievable via `transform-runtime` but if that is not the case I'd love to talk through what is missing.

**Comment by maniax89** [comment]
I added a comment over in https://github.com/babel/babel/issues/5699#issuecomment-352105081 with a use-case that I don't think is achievable with `transform-runtime`
